src/Microsoft.Data/BinaryOperations/DataFrameBinaryOperations.tt

```diff
@@ -0,0 +1,69 @@
+﻿<#@ template debug="false" hostspecific="false" language="C#" #>
+<#@ assembly name="System.Core" #>
+<#@ import namespace="System.Linq" #>
+<#@ import namespace="System.Text" #>
+<#@ import namespace="System.Collections.Generic" #>
+<#@ output extension=".cs" #>
+<#@ include file="..\DataFrameColumnArithmeticTemplate.ttinclude" #>
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Data
+{
+    public partial class DataFrame
+    {
+        #region Binary Operations
+
+<# foreach (MethodConfiguration method in methodConfiguration) { #>
+<# if (method.MethodType == MethodType.BinaryScalar || method.MethodType == MethodType.ComparisonScalar) {#>
+        public DataFrame <#=method.MethodName#><T>(T value)
+<# } else if (method.MethodType == MethodType.BinaryInt) { #>
+        public DataFrame <#=method.MethodName#><T>(int value)
+<# } else { #>
+        public DataFrame <#=method.MethodName#><T>(IList<T> values)
+<# } #>
+            where T : struct
+        {
+<# if (method.MethodType == MethodType.BinaryScalar || method.MethodType == MethodType.ComparisonScalar || method.MethodType == MethodType.BinaryInt) {#>
+<# } else { #>
+            if (values.Count != NumColumns)
+            {
+                throw new ArgumentException($"values.Count {values.Count} must match the number of columns in the table", nameof(values));
+            }
+<# } #>
+            var newDataFrame = new DataFrame();
+            for (int ii = 0; ii < NumColumns; ii++)
+            {
+                DataFrameColumn<T> column = _table.Column(ii) as DataFrameColumn<T>;
+                if (column != null)
```

Comment:
I'm not opposed to the style but I do have some questions here for the larger crowd:

If I had a `DataFrameColumn<int>` and called `DataFrameColumn<int>.Add(5.0)`,
The lines 
`DataFrameColumn<T> column = _table.Column(ii) as DataFrameColumn<T>; 
                if (column != null)`

mean that nothing will happen since the dynamic cast will fail. I'm leaning towards checking the type of the resulting operation and creating a new column with that type?
For ex: DataFrameColumn<int>.Add(5.0) will result in a new DataFrameColumn<double>, similar to what `1+5.0` does in C#.

For cases such as DataFrameColumn<int> + string, we just don't perform the operation instead of throwing?
Thoughts?

Reply:
I would expect it to throw if I try an operation that isn't supported, but why wouldn't that one follow the same rule as number + string with scalars? i.e. convert to string

Reply:
Ugh, my bad. I typed in a hurry. I meant to write "DataFrameColumn - string"(Basically whenever we have incompatible operands). Edited it now.